src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/types.hpp

```diff
@@ -657,19 +657,172 @@ struct b_maxkey {
 ///
 /// free function comparator for b_maxkey
 ///
-/// @relatesalso b_maxkey
+/// @relatesalso bsoncxx::v_noabi::types::b_maxkey
 ///
 BSONCXX_INLINE bool operator==(const b_maxkey&, const b_maxkey&) {
     return true;
 }
 
+#if defined(BSONCXX_PRIVATE_DOXYGEN_PREPROCESSOR)
```

Comment:
Instead of using `BSONCXX_PRIVATE_DOXYGEN_PREPROCESSOR`, consider defining the functions and removing the use of the `BSONCXX_ENUM` X-macro:

```c++
///
/// @relatesalso bsoncxx::v_noabi::types::b_double
///
/// free function comparator for b_double
///
BSONCXX_INLINE bool operator!=(const b_double& lhs, const b_double& rhs) {
    return !(lhs == rhs);
}
// And others ...
```

That may help avoid inconsistency between the documentation and the used declarations (e.g. the documented types do not currently include `BSONCXX_INLINE`).

Reply:
Good call. Done as suggested.